monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/BasePlugins.js

```diff
@@ -12,42 +12,22 @@ const initialState = {
 };
 
 const BasePlugins = (props) => {
-  const {plugins, setPlugins, getRowActions, onRefreshCallback, loadingMessage = DEFAULT_LOADING_MESSAGE} = {...props};
-
-  const [isLoadingPlugins, setIsLoadingPlugins] = useState(false);
-
-  useEffect(() => {
-    setPluginsList();
-  }, []);
+  const {plugins, getRowActions, onRefreshCallback, loadingMessage = DEFAULT_LOADING_MESSAGE} = {...props};
 
   const rows = useMemo(() => {
     return getPluginsGridRows(plugins);
   }, [plugins]);
 
-  const setPluginsList = () => {
-    if(plugins.length === 0) {
-      setIsLoadingPlugins(true);
-      getPlugins(false, false, true).then(plugins => {
-        setPlugins(plugins);
-      }).finally(() => {
-        setIsLoadingPlugins(false);
-      });
-    }
-  };
-
   // eslint-disable-next-line no-unused-vars
   const onRefreshClick = () => {
-    setPluginsList();
-    if(onRefreshCallback) {
-      onRefreshCallback();
-    }
+    onRefreshCallback();
```

Comment:
If there isn't any onRefreshCallback prop, it will try to execute the function although it's null. That will raise an exception during the run.
We need to keep the check that the prop is actually available.